docs/docs/sourcing-from-flotiq.md

```diff
@@ -0,0 +1,182 @@
+
+   ```shell
+   cd my-blog-starter/
+   npm install
+   gatsby develop
+   ```
+
+   If you wish to import example blog posts to your account, before running `gatsby develop` run:
+
+   ```shell
+   node ./.flotiq/importExample.js
+   ```
+
+   It will add 1 image and 2 blog posts to your Flotiq account.
+
+   _Note: You need to put your Read and write API key in `.env` for import to work. You don't need Blog Post content type in your account. If you already have posts with ids `blogpost-1` and `blogpost-2` they will be overwritten._
+
+### Source plugin
+
+The Blog starter you've just set up uses the [gatsby-source-flotiq](https://github.com/flotiq/gatsby-source-flotiq) source plugin to pull content from Flotiq.
+
+Here are the steps to use this source plugin in other Gatsby projects:
+
+> If you're using the provided starter - all the following steps have been already taken care of, you can dive into the relevant project files to verify how it's been done.
+
+1. Install the plugin:
+   ```shell
+   npm install --save gatsby-source-flotiq
+   ```
+2. Provide API credentials in `.env`:
+3. Register the source plugin in `gatsby-config.js`:
+   Make sure your `gatsby-config.js` has the following configuration:
+   ```javascript:title=gatsby-config.js
+   module.exports = {
+     // ...
+     plugins: [
+       {
+         resolve: "gatsby-source-flotiq",
+         options: {
+           baseUrl: process.env.GATSBY_FLOTIQ_BASE_URL,
+           authToken: process.env.FLOTIQ_API_KEY,
+         },
+       },
+     ],
+     // ...
+   }
+   ```
+
+By default the source plugin will pull all the data you've stored in Flotiq. In some cases, e.g. when building sites that have thousands of pages - you'll likely want to limit the number of pulled data during development. You can do that either by narrowing down the scope to specific content types or by limiting the number of pulled objects (see the `includeTypes` and `objectLimit` options). The source plugin will do its best to resolve any dependencies for you.
+
+> You can find more details about how to configure the Flotiq source plugin in the [plugin's README](https://github.com/flotiq/gatsby-source-flotiq).
+
+## Accessing Flotiq data in Gatsby
+
+Once you finish configuring your environment you can start developing. The source plugin will pull all the content from your Flotiq account and create respective Gatsby GraphQL nodes. It will also preserve the relations you've setup in Flotiq, so the GraphQL nodes will be automatically linked, just as you would expect.
+
+For example, if you define a relation between `BlogPost` and `Category` content types in Flotiq - they will retain their relationship in Gatsby's GraphQL, so you can retrieve that in a query:
+
+```graphql
+query {
+  allBlogPost {
+    nodes {
+      id
+      slug
+      category {
+        name
+      }
+    }
+  }
+}
+```
+
+The next step is implementing these queries in your `gatsby-node.js`
+
+```javascript:title=gatsby-node.js
```

Comment:
This example doesn't include all the required source code, so it could really trip people up trying to follow the guide–it did for me. Can you add a more complete example along with a simplified blog post template? I'm also getting an error when I try to follow the instructions using the Gatsby default starter, even with an `.env` file at the root of the project it throws:

```
➜  flotiq-demo-scratch git:(master) ✗ gatsby develop
success open and validate gatsby-configs - 0.023s
success load plugins - 0.419s
success onPreInit - 0.013s
success initialize cache - 0.016s
success copy gatsby files - 0.051s
success onPreBootstrap - 0.007s
success createSchemaCustomization - 0.006s

 ERROR

FLOTIQ: You must specify API url (in most cases it is "https://api.flotiq.com")
```

Reply:
I apologize for that, it was probably because of the improper name used in the config snippet. I pushed an update that properly references the ENV variable:

```
 {
         resolve: "gatsby-source-flotiq",
         options: {
           baseUrl: process.env.GATSBY_FLOTIQ_BASE_URL,
           authToken: process.env.GATSBY_FLOTIQ_API_KEY,
         },
       },
```

Reply:
I amended the gatsby-node.js file, it now provides a complete working example. I also added a stripped-down blog-post template.